### PR TITLE
Add the TCP_FUNCTION_BLK socket option, on FreeBSD

### PR DESCRIPTION
**Pull request by asomers**
## What does this PR do

Adds the `TCP_FUNCTION_BLK` socket option, used to set or query a socket's TCP stack, on FreeBSD only

## Checklist:

- [x] I have read `CONTRIBUTING.md`
- [x] I have written necessary tests and rustdoc comments
- [x] A change log has been added if this PR modifies nix's API

### CODE REVIEW

**Comment by asomers** [comment]
Note: do not merge until after the libc PR has been merged, and then backported to libc-0.2: https://github.com/rust-lang/libc/pull/4047

File: test/sys/test_sockopt.rs
Comment (SteveLauC) [inline test/sys/test_sockopt.rs:299]: This function has not been stabilized in our MSRV 1.69, we can use:

```suggestion
    assert!(!name.to_bytes().is_empty());
```

**Comment by asomers** [comment]
The relevant PR has been merged to libc, backported to the 0.2 branch, and released in libc 0.2.163.

**Review by SteveLauC — APPROVED** [review]
Thanks!

**Pull request merged**
